tokenserver/assignment/sqlnode/sql.py

```diff
@@ -314,13 +328,35 @@ def get_user(self, service, email):
         finally:
             res.close()
 
+    def get_migration_percent(self):
+        """get a cached Ops controllable percentage value for the number of
+        new users to migrate to Spanner."""
+        if self._cache_migration_ttl > time.time():
+            return self.migrate_new_user_percentage
+        default = self.migrate_new_user_percentage or 0
+        try:
+            res = self._safe_execute(
+                _GET_DYNAMIC_SETTING,
+                {"setting": "migrate_new_user_percentage"})
+            self.migrate_new_user_percentage = (
+                int(res.fetchone()[0]) or default
+            )
+            self._cache_migration_ttl = time.time() + 300
+        except Exception as ex:
+            logger.info(
+                "Could not get migration percent \"{}\" using default: {}"
+                .format(ex, default)
+            )
+        return self.migrate_new_user_percentage
+
     def should_allocate_to_spanner(self, email):
         """use a simple, reproducable hashing mechanism to determine if
         a user should be provisioned to spanner. Does not need to be
         secure, just a selectable percentage."""
-        if self.migrate_new_user_percentage:
+        migrate = self.get_migration_percent()
```

Comment:
When the timer on the cached value expires, we may end up with several concurrent requests trying to read and update the value at once. That's...probably fine i guess? I wonder if it's worth mentioning in a comment here for future readers.

Reply:
I don't think we should see any updates, but it is possible that a few threads have different values while the various TTLs live out. I think that's fine since we're talking about a reasonably short window. Things don't have to be exact, but it'll be faster than a system redeploy.